### ISSUE TITLE
Large batch size and multiple GPUs

### ISSUE THREAD

**Issue by jjcao** [junyanz/pytorch-CycleGAN-and-pix2pix#137]
When I train pix2pix model with batchSize > 1, norm = batch, and multiple GPU, the results seem wrong/strange.

When I train pix2pix model with  batchSize > 1, norm = batch, and single GPU, the results are correct.

Could this be solved?

Thank you.

**Comment by junyanz** [comment]
We observe that batchSize=1 with single gpu gives us the best results so far. 
According to [this post](https://github.com/tensorflow/tensorflow/issues/7439), It seems that pytorch calculates mean/var statistics for each gpu. 
So how many images do you have per GPU? 1 per GPU might cause some problems for batchnorm. 
Have you tried instancenorm with multi-gpu setting and batchSize>1?

**Comment by jjcao** [comment]
Yes.  with --norm instance, it worked.

Specify the number of images per GPU? Is there an option or is it simple for changing your code?

**Comment by junyanz** [comment]
I guess it will be `batchSize/#gpus`.

**Comment by jjcao** [comment]
If it is batchSize/#gpus, then norm still need to be "instance" for successful training. I have tested this.
